includes/breadcrumbs.inc

```diff
@@ -0,0 +1,236 @@
+<?php
+
+/**
+ * @file
+ * This file contains functions to create breadcrumbs using Solr.
+ */
+
+/**
+ * Implements callback_islandora_breadcrumbs_backends.
+ */
+function islandora_solr_islandora_breadcrumbs_backends_callable($object) {
+  $breadcrumbs = islandora_solr_get_breadcrumbs_recursive($object->id);
+  array_pop($breadcrumbs);
+  return $breadcrumbs;
+}
+
+/**
+ * Loads a map of collection hierarchy.
+ *
+ * @return array
+ *   The array has current pid as key pointing to an associative array of
+ *   'parent' => pid and 'label' => parent_label.
+ */
+function islandora_solr_load_crumbs() {
+  if (user_is_anonymous()) {
+    $cache_roles = 'anonymous';
+  }
+  else {
+    global $user;
+    $roles = $user->roles;
+    // Sort so we keep ordering consistent to reuse cache.
+    asort($roles);
+    $cache_roles = implode('_', $roles);
+  }
+
+  $crumb_cache = &drupal_static(__FUNCTION__);
+
+  $breadcrumbs = array();
+
+  if (!isset($crumb_cache)) {
+    $cache = cache_get(ISLANDORA_SOLR_BREADCRUMB_CACHE_NAME, 'cache_menu');
+    if (!$cache) {
+      // Nothing in cache so initialize it with these roles.
+      $fetch = $cache_roles;
+    }
+    else {
+      $crumb_cache = $cache->data;
+    }
+  }
+
+  if (!isset($crumb_cache[$cache_roles])) {
+    // No cache for this set of roles.
+    $fetch = $cache_roles;
+  }
+  else {
+    $breadcrumbs = $crumb_cache[$cache_roles];
+  }
+
+  if (!$breadcrumbs) {
+    $fetch = $cache_roles;
+  }
+
+  if (isset($fetch)) {
+    module_load_include('inc', 'islandora_solr_search', 'includes/query_processor');
+    $solr_build = new IslandoraSolrQueryProcessor();
+
+    $content_model = variable_get('islandora_solr_content_model_field', 'RELS_EXT_hasModel_uri_ms');
+    $collection_field = variable_get('islandora_solr_member_of_collection_field', 'RELS_EXT_isMemberOfCollection_uri_ms');
```

Comment:
So I could modify the [islandora_solr_get_parents_from_rels_ext](https://github.com/whikloj/islandora_solr_search/blob/7.x-ISLANDORA-1300/includes/breadcrumbs.inc#L221-L236) to only grab relationships that match the set created by re-using the [hook_islandora_get_breadcrumb_query_predicates](https://github.com/Islandora/islandora/blob/7.x/islandora.api.php#L867-L872).

But again, the method I'm using only cares about isMemberOf and isPartOf until you hit a islandora:collectionCModel. Then it is assuming isMemberOfCollection all the way to the root.

If that is not good, let me know and I can look at a straight query for each object. 

But that will require knowing all the Solr fields one needs which can't be guessed from the existing hook. So we need a) a new hook b) a new configuration variable.